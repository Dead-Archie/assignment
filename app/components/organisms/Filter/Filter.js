/* eslint-disable react/no-unused-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import FilterComponent from '../../molecules/FilterComponent';

class Filter extends React.Component {
  getSpecies = () => {
    const { data } = this.props;
    const speciesObj = {};
    // eslint-disable-next-line array-callback-return
    data.map(value => {
      const { species } = value;
      if (!speciesObj[species]) {
        speciesObj[species] = '';
      }
    });
    return Object.keys(speciesObj);
  };

  getOrigin = () => {
    const { data } = this.props;
    const originObj = {};
    // eslint-disable-next-line array-callback-return
    data.map(value => {
      const {
        origin: { name },
      } = value;
      if (!originObj[name]) {
        originObj[name] = '';
      }
    });
    return Object.keys(originObj);
  };

  render() {
    const { data } = this.props;
    if (!data) return null;
    const speciesMap = this.getSpecies();
    const originMap = this.getOrigin();
    const { query } = this.props;
    return (
      <div>
        <h3>Filters</h3>
        {speciesMap && <FilterComponent data={speciesMap} title="species" query={query} />}
        <FilterComponent data={['male', 'female']} title="gender" query={query} />
        {originMap && <FilterComponent data={originMap} title="origin" query={query} />}
      </div>
    );
  }
}

Filter.propTypes = {
  data: PropTypes.shape([]),
  query: PropTypes.shape({}),
};
Filter.defaultProps = {
  data: [],
  query: {},
};

export default Filter;
export { Filter };
