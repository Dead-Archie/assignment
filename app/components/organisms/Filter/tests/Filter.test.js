import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Filter } from '../Filter';

const sampleProps = {
  className: 'withStyles-sc-4wveuc-0 lgzevM',
  query: {
    name: 'morty',
    origin: 'Earth (C-137)',
  },
  data: [
    {
      gender: 'male',
      name: 'Morty Smith',
      origin: {
        name: 'Earth (C-137)',
        url: 'https://rickandmortyapi.com/api/location/1',
      },
      species: 'Human',
      status: 'Alive',
      type: '',
      location: {
        name: 'Earth (Replacement Dimension)',
        url: 'https://rickandmortyapi.com/api/location/20',
      },
    },
  ],
};

const setUp = props => {
  return shallow(<Filter {...props} />);
};

describe('<Filter />', () => {
  let FilterComponent;
  beforeEach(() => {
    FilterComponent = setUp(sampleProps);
  });

  test('should render with out props correctly', () => {
    const wrapper = shallow(<Filter data={null} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should render correctly', () => {
    expect(FilterComponent).toMatchSnapshot();
  });
});
