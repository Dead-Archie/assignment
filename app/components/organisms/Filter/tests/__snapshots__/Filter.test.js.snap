// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render correctly 1`] = `
<div>
  <h3>
    Filters
  </h3>
  <FilterComponent
    data={
      Array [
        "Human",
      ]
    }
    query={
      Object {
        "name": "morty",
        "origin": "Earth (C-137)",
      }
    }
    title="species"
  />
  <FilterComponent
    data={
      Array [
        "male",
        "female",
      ]
    }
    query={
      Object {
        "name": "morty",
        "origin": "Earth (C-137)",
      }
    }
    title="gender"
  />
  <FilterComponent
    data={
      Array [
        "Earth (C-137)",
      ]
    }
    query={
      Object {
        "name": "morty",
        "origin": "Earth (C-137)",
      }
    }
    title="origin"
  />
</div>
`;

exports[`<Filter /> should render with out props correctly 1`] = `""`;
